src/operator/controllers/intents_reconcilers/network_policy.go

```diff
@@ -317,6 +328,66 @@ func (r *NetworkPolicyReconciler) handleNetworkPolicyRemoval(
 	return nil
 }
 
+func (r *NetworkPolicyReconciler) removeOrphanNetworkPolicies(ctx context.Context) error {
+	logrus.Info("Searching for orphaned network policies")
+	networkPolicyList := &v1.NetworkPolicyList{}
+	selector, err := r.matchAccessNetworkPolicy()
```

Comment:
We can index NetworkPolicies by its "from namespace selector". This way we will be able to list only policies that were referenced by intents from the reconciled intent's namespace